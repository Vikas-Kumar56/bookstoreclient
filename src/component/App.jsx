import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Layout from './layout/Layout';
import BookContainer from './book/BookContainer';
import Login from './user/Login';
import { SnackbarProvider } from 'notistack';
import Auth from './Auth';

const App = () => {
  return (
    <SnackbarProvider maxSnack={3}>
      <Router>
        <Layout>
          <Switch>
            <Route path='/login'>
              <Login />
            </Route>
            <Route exact path='/'>
              <Auth>
                <BookContainer />
              </Auth>
            </Route>
          </Switch>
        </Layout>
      </Router>
    </SnackbarProvider>
  );
};

export default App;
