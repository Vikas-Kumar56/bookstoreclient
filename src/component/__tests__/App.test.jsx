import React from "react";
import renderWithRedux from "../../util/testUtil";
import App from "../App";

describe("App component", () => {
  it("should render app with error", () => {
    const { asFragment } = renderWithRedux(<App />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
