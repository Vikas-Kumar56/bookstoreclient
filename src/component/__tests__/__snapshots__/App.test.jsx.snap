// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app with error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Book Store
          </h6>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root MuiBox-root-77"
    >
      <div
        class="MuiBox-root MuiBox-root-84 makeStyles-bookContainer-78"
      >
        <div
          class="MuiBox-root MuiBox-root-85 makeStyles-bookFilter-79"
        >
          <div
            class="MuiPaper-root makeStyles-bookFilterPaper-80 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Search Book Filters
            </p>
            <div
              class="MuiBox-root MuiBox-root-86"
            >
              <div
                class="MuiFormControl-root MuiTextField-root"
                data-testid="book-title-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  for="book-search"
                  id="book-search-label"
                >
                  Enter book title
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="book-search"
                    placeholder="Enter book title"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-147"
                    >
                      <span>
                        Enter book title
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-181 makeStyles-bookList-81"
        >
          <div
            class="MuiBox-root MuiBox-root-192"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              data-testid="book-loader"
              style="width: 80%; height: 200px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
